packages/react/src/components/DataTable/TableToolbarSearch.tsx

```diff
@@ -336,14 +337,7 @@ TableToolbarSearch.propTypes = {
   /**
    * Provide custom text for the component for each translation id
    */
-  translateWithId: PropTypes.func.isRequired,
-};
-
-TableToolbarSearch.defaultProps = {
-  tabIndex: '0',
-  translateWithId,
-  persistent: false,
-  onClear: () => {},
+  translateWithId: PropTypes.func,
```

Comment:
Removing `isRequired`, as we do not explicitly require a user to pass this prop into the component anywhere else in the repo and a reasonable default is provided.